src/caches/basic_caches.jl

```diff
@@ -22,21 +22,21 @@ function alg_cache(alg::CompositeAlgorithm{Tuple{T1, T2}, F}, u, rate_prototype,
     uprev2, f, t, dt, reltol, p, calck,
     ::Val{V}) where {T1, T2, F, V, uEltypeNoUnits, uBottomEltypeNoUnits,
     tTypeNoUnits}
-    caches = __alg_cache(alg.algs, u, rate_prototype, uEltypeNoUnits, uBottomEltypeNoUnits,
-        tTypeNoUnits, uprev, uprev2, f, t, dt, reltol, p, calck, Val(V))
+    caches = (alg_cache(alg.algs[1], u, rate_prototype, uEltypeNoUnits,
+            uBottomEltypeNoUnits,
+            tTypeNoUnits, uprev, uprev2, f, t, dt, reltol, p, calck, Val(V)),
+        alg_cache(alg.algs[2], u, rate_prototype, uEltypeNoUnits,
+            uBottomEltypeNoUnits,
+            tTypeNoUnits, uprev, uprev2, f, t, dt, reltol, p, calck, Val(V)))
```

Comment:
Not all compositecaches are only two, make this branch on whether it's a two tuple

Reply:
oh I see this is already a special dispatch